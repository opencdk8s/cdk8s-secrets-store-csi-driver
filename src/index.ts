import { ApiObject, GroupVersionKind } from 'cdk8s';
import { Construct } from 'constructs';
import * as k8s from './imports/k8s';
export * as k8s from './imports/k8s';

export interface SecretObjectData {
  readonly objectName?: string;
  readonly key?: string;
}

export interface SecretObject {
  readonly secretName?: string;
  readonly type?: string;
  readonly labels?: { [key: string]: string};
  readonly annotations?: { [key: string]: string};
  readonly data?: SecretObjectData[];
}

export enum Provider {
  AWS = 'aws',
  VAULT = 'vault',
  AZURE = 'azure',
  GCP = 'gcp'
}

export interface SecretProviderClassSpec {
  readonly provider?: Provider;
  readonly parameters?: { [key: string]: any};
  readonly secretObjects?: SecretObject[];
}

export interface ByPodStatus {
  readonly id?: string;
  readonly namespace?: string;
}

export interface SecretProviderClassStatus {
  readonly byPod?: ByPodStatus[];
}

export interface SecretProviderClassProps {
  /**
   * Standard object's metadata.
   *
   */
  readonly metadata?: k8s.ObjectMeta;
  readonly spec?: SecretProviderClassSpec;
  readonly status?: SecretProviderClassStatus;
}

export interface SecretProviderClassListProps {
  readonly meta?: k8s.ListMeta;
  readonly items?: SecretProviderClassProps[];
}

export class SecretProviderClass extends ApiObject {
  public static readonly GVK: GroupVersionKind = {
    apiVersion: 'secrets-store.csi.x-k8s.io/v1',
    kind: 'SecretProviderClass',
  };
  /**
   * @param props initialization props
   */
  public static manifest(props: SecretProviderClassProps = {}): any {
    return {
      ...SecretProviderClass.GVK,
      ...props,
    };
  }

  /**
   * @param scope the scope in which to define this object
   * @param id a scope-local name for the object
   * @param props initialization props
   */
  public constructor(scope: Construct, id: string, props: SecretProviderClassProps = {}) {
    super(scope, id, SecretProviderClass.manifest(props));
  }
}

export class SecretProviderClassList extends ApiObject {
  public static readonly GVK: GroupVersionKind = {
    apiVersion: 'secrets-store.csi.x-k8s.io/v1',
    kind: 'SecretProviderClassList',
  };
  /**
   * @param props initialization props
   */
  public static manifest(props: SecretProviderClassListProps = {}): any {
    return {
      ...SecretProviderClassList.GVK,
      ...props,
    };
  }

  /**
   * @param scope the scope in which to define this object
   * @param id a scope-local name for the object
   * @param props initialization props
   */
  public constructor(scope: Construct, id: string, props: SecretProviderClassListProps = {}) {
    super(scope, id, SecretProviderClassList.manifest(props));
  }
}
