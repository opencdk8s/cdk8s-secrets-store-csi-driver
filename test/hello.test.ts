import { Testing, Chart } from 'cdk8s';
import * as t from '../src/index';
test('hello', () => {
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  new t.SecretProviderClass(chart, 'test1');
  new t.SecretProviderClassList(chart, 'test');


  expect(Testing.synth(chart)).toMatchSnapshot();
});
