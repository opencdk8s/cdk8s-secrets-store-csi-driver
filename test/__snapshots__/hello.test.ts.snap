// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello 1`] = `
Array [
  Object {
    "apiVersion": "secrets-store.csi.x-k8s.io/v1",
    "kind": "SecretProviderClass",
    "metadata": Object {
      "name": "test-test1-c8bcf0bf",
    },
  },
  Object {
    "apiVersion": "secrets-store.csi.x-k8s.io/v1",
    "kind": "SecretProviderClassList",
    "metadata": Object {
      "name": "test-c87b5978",
    },
  },
]
`;
